packages/scenes/src/behaviors/CursorSync.ts

```diff
@@ -0,0 +1,77 @@
+import { BusEvent, BusEventHandler, BusEventType, EventBus, EventFilterOptions } from '@grafana/data';
+import { DashboardCursorSync } from '@grafana/schema';
+import { filter, Observable, Unsubscribable } from 'rxjs';
+import { sceneGraph } from '../core/sceneGraph';
+import { SceneObjectBase } from '../core/SceneObjectBase';
+import { SceneObject, SceneObjectState } from '../core/types';
+
+interface CursorSyncState extends SceneObjectState {
+  sync: DashboardCursorSync;
+}
+
+/**
+ * This behavior will provide a cursor sync context within a scene.
+ */
+
+export class CursorSync extends SceneObjectBase<CursorSyncState> {
+  public constructor(state: Partial<CursorSyncState>) {
+    super({
+      ...state,
+      sync: state.sync || DashboardCursorSync.Off,
+    });
+  }
+
+  public getEventsBus = (panel: SceneObject) => {
+    if (!this.parent) {
+      throw new Error('EnableCursorSync cannot be used as a standalone scene object');
+    }
+    // Since EnableCursorSync is a behavior, it is not a parent to any object in the scene graph.
+    // We need to get it's parent in order to provide correct EventBus context to the children.
+    return new PanelContextEventBus(this.parent, panel);
+  };
+
+  public getEventsScope = () => {
+    if (!this.parent) {
+      throw new Error('EnableCursorSync cannot be used as a standalone scene object');
+    }
+    // Since EnableCursorSync is a behavior, it is not a parent to any object in the scene graph.
+    // We need to get it's parent in order to provide correct EventBus context to the children.
+    return this.parent.state.key;
+  };
+}
+
+// This serves as a shared EventsBus that is shared by children or CursorSync behavior.
+class PanelContextEventBus implements EventBus {
+  public constructor(private _source: SceneObject, private _eventsOrigin: SceneObject) {}
+
+  public publish<T extends BusEvent>(event: T): void {
+    (event as any).origin = this._eventsOrigin;
```

Comment:
```suggestion
    (event as any).origin = this;
```

I think origin needs to be the bus, as that is what we GraphNG is comparing against, but since that filtering is done internally here I guess we can decide on making origin something else

Reply:
Yeah, with this change the filtering in GraphNG is never called. I'll open a separate work that will address that in the core Grafana.